server/src/test/java/e2e/RescueThreadRunTest.java

```diff
@@ -83,7 +86,14 @@ void shouldExecuteFailedNodeAgainOnRescue() {
                 .waitForNodeRunStatus(0, 1, LHStatus.ERROR)
                 .thenRescueThreadRun(0, false) // dont skip failed node: try again
                 .waitForStatus(LHStatus.COMPLETED)
-                .thenVerifyAllTaskRuns(0, taskRuns -> {
+                .thenVerifyAllTaskRuns(unsortedTaskRuns -> {
```

Comment:
In my opinion, this is not semantically correct and goes against the objectives of this test framework, which is to provide a way to describe a set of verifiable individual constraints that enable clients (server devs)  describe how the workflow should look after applying some inputs (external events, input variables, user task forms, etc.). What I see in the `thenVerifyAllTaskRuns` method is a way to break that promise, because this allows to include a big step that executes a lot of assertions in it. My suggestion here is to include separate steps with fixed node positions to verify each task run.

Reply:
Can we make this change in another PR?

Reply:
Sure, we can create a separate PR for this.